pages/Documents/ConversationalAI/Third-PartyBots/MicrosoftDirectLine/advanced-features.md

```diff
@@ -0,0 +1,325 @@
+
+A single private text message with an action can be sent by adding `text` and `messageAudience` properties with 
+relevant action (e.g. [Transfer](third-party-bots-microsoft-direct-line-basic-content.html#transfer)) 
+properties. An example of such case is below:
+
+```json-doc
+{
+  // ...
+  "type": "message",
+  "text": "This is a private text",
+  "channelData": {
+    "messageAudience": "AGENTS_AND_MANAGERS",
+    "action": {
+      "name": "TRANSFER",
+      "type": "client",
+      "parameters": {
+        "skill": "human_skill"
+      },
+      "result_variable": "none"
+    }
+  }
+}
+```
+
+Figure 2.1 Transfer activity with a private message visible to agents and managers
+
+### Engagement attributes as context
+
+Third-Party bots allows the collection of engagement attributes (more information can be found 
+[here](engagement-attributes-types-of-engagement-attributes.html)) if the `Engagement Attributes` option is checked in 
+the `Conversation Type` step as shown in Figure 3.1.
+
+<img class="fancyimage" style="width:750px" src="img/engagement_attr_select.png">
+Figure 3.1 Conversation Type step in creation/modification of bot configuration.
+
+These attributes are **only** collected at the start of a conversation. Third-Party bots leverage the LivePerson 
+Visit Information API to collect the engagement attributes. Further information on the Visit Information API 
+can be found [here](visit-information-api-visit-information.html). Moreover, engagement attributes are not updated 
+throughout the life cycle of a conversation and only passed along with each message request. 
+In Microsoft Bot these engagement attributes are added to the property `lpSdes` which is part of another custom 
+property of `context`. This context information within a conversation is preserved/passed in the `channelData` 
+property (further information about `channelData` can be found 
+[here](https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-api-reference?view=azure-bot-service-4.0#activity-object)
+). An example of the activity can be seen below:
+
+```json-doc
+{
+  // ...
+  "type": "message",
+  "text": "Some Conversation Text",
+  "channelData": {
+    "context": {
+      "lpEvent": {}, // Holds LP Events
+      "lpSdes": {}
+    }
+  }
+}
+```
+
+Figure 3.1 Customer activity excerpt on a new chat
+
+### Sending Multiple Responses
+
+As stated under Limitations the default behaviour of our connector is to process the first responses we find on the channel. In case `Multiple Activities`
+is not enabled or the waiting period is set to low, the connector might not retrieve every activity your bot is sending.
+
+In that case we also provide a way to define multiple bot responses in a single Direct Line activity. As with all channel specific content this is defined in the channelData property.
+The array in the multiMessage property can contain the objects identified by the following types:
+
+- `text A plain message
```

Comment:
I think here are some format problems right?? text looks like not closed with `

Reply:
Thanks, fixed it now